torchmultimodal/models/video_vqvae.py

```diff
@@ -7,11 +7,87 @@
 from typing import Any, Dict, List, Tuple, Union
 
 from torch import nn, Tensor
-from torchmultimodal.modules.layers.attention import AxialAttentionBlock
 
+from torchmultimodal.models.vqvae import VQVAE
+from torchmultimodal.modules.layers.attention import AxialAttentionBlock
 from torchmultimodal.modules.layers.conv import SamePadConv3d, SamePadConvTranspose3d
 from torchmultimodal.utils.common import format_convnet_params
 
+DEFAULT_ENCODER_IN_CHANNEL_DIMS = (3, 240, 240, 240, 240, 240)
+DEFAULT_ENCODER_STRIDES = (
+    (2, 2, 2),
+    (2, 2, 2),
+    (1, 2, 2),
+    (1, 2, 2),
+    (1, 2, 2),
+    (1, 1, 1),
+)
+DEFAULT_DECODER_OUT_CHANNEL_DIMS = (240, 240, 240, 240, 3)
+DEFAULT_DECODER_STRIDES = (
+    (2, 2, 2),
+    (2, 2, 2),
+    (1, 2, 2),
+    (1, 2, 2),
+    (1, 2, 2),
+)
+
+
+def video_vqvae(
+    encoder_in_channel_dims: Tuple[int, ...] = DEFAULT_ENCODER_IN_CHANNEL_DIMS,
+    encoder_kernel_sizes: Union[int, Tuple[int, ...], Tuple[Tuple[int, ...], ...]] = 3,
+    encoder_strides: Union[
+        int, Tuple[int, ...], Tuple[Tuple[int, ...], ...]
+    ] = DEFAULT_ENCODER_STRIDES,
+    n_res_layers: int = 4,
+    attn_hidden_dim: int = 240,
+    num_embeddings: int = 2048,
+    embedding_dim: int = 256,
+    decoder_out_channel_dims: Tuple[int, ...] = DEFAULT_DECODER_OUT_CHANNEL_DIMS,
+    decoder_kernel_sizes: Union[int, Tuple[int, ...], Tuple[Tuple[int, ...], ...]] = 3,
+    decoder_strides: Union[
+        int, Tuple[int, ...], Tuple[Tuple[int, ...], ...]
+    ] = DEFAULT_DECODER_STRIDES,
```

Comment:
I like the generality but wonder if we should impose some constraints just to make the signature a little cleaner here

Reply:
unfortunately the default stride for MUGEN uses the most configurable type (tuple of tuples) so I had to include all three types in the union here. some ideas to make this cleaner are to call this whole union a type above the function and then use that just to make it cleaner: `conv_param_type = Union[int, Tuple[int, ...], Tuple[Tuple[int, ...], ...]]`. Or we don't overfit to MUGEN's defaults and use a very general type, like `int` only or `Tuple` only.